apps/studio/src/components/sidebar/core/table_list/VirtualTableList.vue

```diff
@@ -0,0 +1,195 @@
+<template>
+  <virtual-list
+    ref="vList"
+    class="list-body"
+    :data-key="'key'"
+    :data-sources="displayItems"
+    :data-component="itemComponent"
+    :estimate-size="itemHeight"
+    :keeps="keeps"
+    :extra-props="{ onExpand: handleExpand, onPin: handlePin }"
+  ></virtual-list>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import { TableOrView, Routine } from "@/lib/db/models";
+import TableListContextMenus from "@/mixins/TableListContextMenus";
+import ItemComponent from "./Item.vue";
+import VirtualList from "vue-virtual-scroll-list";
+import { AppEvent } from "@/common/AppEvent";
+import { mapGetters, mapState } from "vuex";
+import { PinnedEntity } from "@/common/appdb/models/PinnedEntity";
+
+type Entity = TableOrView | Routine | string;
+
+type Item = SchemaItem | TableItem;
+
+interface BaseItem {
+  type: "schema" | "table" | "routine";
+  entity: Entity;
+  key: string;
+  expanded: boolean;
+  hidden: boolean;
+  contextMenu: any[];
+  level: number;
+}
+
+interface SchemaItem extends BaseItem {
+  type: "schema";
+  entity: string;
+  skipDisplay: boolean;
+}
+
+interface TableItem extends BaseItem {
+  type: "table" | "routine";
+  entity: TableOrView | Routine;
+  parent: SchemaItem;
+  pinned: boolean;
+}
+
+export default Vue.extend({
```

Comment:
This looks awesome, good stuff! Can you add a nice verbose comment to this file explaining why we need it?

Pretend someone new is starting on the team in 12 months, and they're reading the code for the first time -- what information is useful for them to know?

Reply:
Will do :)